New Object management endpoint added

danielzumb opened the pull request.

danielzumb committed:
Added ObjectManagement endpoint base

danielzumb committed:
no message

tiagosiebler commented on lib/rest.js:
```diff
@@ -27,6 +28,9 @@ module.exports = class MicroStrategyRESTAPI extends RestConnection {
     this.reports = new ReportsAPI(this);
 
+    // https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Object%20Management
+    this.objectManagement = new ObjectManagementAPI(this);
```
I missed the mark here in the user management PR, but I meant to keep alphabetical order here (same as the api-docs page). Can you move objectManagement and userManagement to the right positions? It's minor now but can quickly get messy while we add the other endpoints.

tiagosiebler commented on lib/restAPIs/objectManagement.js:
```diff
@@ -0,0 +1,100 @@
+      limit: limit
+    }
+    if (fields) {
+      queryParameter.fields = fields;
```
I think it's worth waiting for #9 to merge then leveraging that fn for this trio of repeated parameters.

tiagosiebler commented on samples/objectManagement/modifyObjects.js:
```diff
@@ -0,0 +1,48 @@
+  const objectID = '6DBBD2404562BC0C580E1B8B18DAEAA8';
+  const type = 8; // Integer from EnumDSSXMLObjectTypes
+
+  const copyObjecctBody = {
```
Minor but typo in var name.

tiagosiebler reviewed (changes requested):
Looks good, just some minor remarks.

danielzumb committed:
Merge branch 'master' of https://github.com/tiagosiebler/microstrategy-api-node into objectManagement

danielzumb committed:
Updated Object Management to use new buildCommonQueryParams function

Updated rest.js to be alphabetical
Fixed type in sample

Testing sample with fields to ensure it gets appended correctly

tiagosiebler commented on lib/restAPIs/objectManagement.js:
```diff
@@ -0,0 +1,83 @@
+    .then(result => this.throwIfFailed(result, 200));
+  }
+
+  recommendObject(objectId, offset, limit, fields){
```
I think this should be `getRecommendations`

tiagosiebler replied:
https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Object%20Management/getRecommendations

danielzumb replied:
The curl URL when you test it on the page is:

`https://demo.microstrategy.com/MicroStrategyLibrary/api/objects/12345/**recommendations**?offset=0&limit=-1`

which just uses recommendations. Also threw it in a sample to test and it gives me recommendations back for a dossier so I think it's all good

tiagosiebler replied:
The prefix denotes the purpose. This API doesn't set an object as recommended, it GETs a list of recommendations based on the query. From my perspective "recommend" seems a bit like "setRecommended" - thought "getRecommendations" would be more intuitive & self documenting from the fn name alone.

danielzumb replied:
Ah I see what you were saying. Thought you meant the API endpoint, yeah that name makes sense, i've updated it

danielzumb reviewed (commented):
The getRecommendations endpoint URL is actually just /recommendations/. Tested it in a sample and check the curl of the docs page

danielzumb committed:
Updated name for recommendations

tiagosiebler reviewed (approved):
👍
